internal/dag/gatewayapi_processor.go

```diff
@@ -52,7 +52,7 @@ type matchConditions struct {
 	headerMatchCondition []HeaderMatchCondition
 }
 
-// Run translates Service APIs into DAG objects and
+// Run translates Gateway API objects into DAG objects and
 // adds them to the DAG.
 func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
```

Comment:
> The gateway didn't have to exist, only be configured in Contour. 

@stevesloka we must have a bug because I see the error when gateway is configured but the gateway does not exist:
```
$ go run ./cmd/contour serve --disable-leader-election --insecure --xds-address=0.0.0.0 --config-path=./config/contour.yaml
INFO[0000] args: [serve --disable-leader-election --insecure --xds-address=0.0.0.0 --config-path=./config/contour.yaml] 
INFO[0000] Leader election disabled                     
...
ERRO[0000] Gateway is not defined!                       context=GatewayAPIProcessor
```

The error goes away when I create the projectcontour/contour gateway.

Reply:
The message probably needs updated, but essentially you're trying to process resources without the gateway existing, so the processor throws that error. So configuring which gateway to use is step #1, then actually having the resource in the cluster is step #2.

Reply:
> you're trying to process resources without the gateway existing

Understood, this is why the `p.source.gateway == nil` check needs to be removed from the gatewayapi processor. This is now a moot point for this PR since the scope strictly focuses on the gatewayclass controller.